pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -315,22 +318,36 @@ func (_ OutboundProxyGenerator) determineRoutes(proxy *model.Proxy, outbound *ku
 		return clusters
 	}
 
-	for _, http := range route.Spec.GetConf().GetHttp() {
-		route := envoy_common.Route{
-			Match:    http.Match,
-			Modify:   http.Modify,
-			Clusters: clustersFromSplit(http.GetSplitWithDestination()),
+	addRoutes := func(route *mesh_core.TrafficRouteResource, rateLimit *mesh_proto.RateLimit, routes envoy_common.Routes) envoy_common.Routes {
+		for _, http := range route.Spec.GetConf().GetHttp() {
+			route := envoy_common.Route{
+				Match:     http.Match,
+				Modify:    http.Modify,
+				RateLimit: rateLimit,
+				Clusters:  clustersFromSplit(http.GetSplitWithDestination()),
+			}
+			routes = append(routes, route)
 		}
-		routes = append(routes, route)
+
+		if defaultDestination := route.Spec.GetConf().GetSplitWithDestination(); len(defaultDestination) > 0 {
+			cfs := clustersFromSplit(defaultDestination)
+			route := envoy_common.Route{
+				Match:     nil,
+				Clusters:  cfs,
+				RateLimit: rateLimit,
+			}
+			routes = append(routes, route)
+		}
+		return routes
 	}
 
-	if defaultDestination := route.Spec.GetConf().GetSplitWithDestination(); len(defaultDestination) > 0 {
-		cfs := clustersFromSplit(defaultDestination)
-		route := envoy_common.Route{
-			Match:    nil,
-			Clusters: cfs,
+	if len(proxy.Policies.RateLimits.Outbound[oface]) < 1 {
+		routes = addRoutes(route, nil, routes)
+	} else {
+		for _, rateLimit := range proxy.Policies.RateLimits.Outbound[oface] {
+			routes = addRoutes(route, rateLimit, routes)
```

Comment:
I don't really follow this. We have one `TrafficRoute` policy for the outbound interface.That policy has HTTP matches that end up generating Envoy routes. If we duplicate the routes with different ratelimit configurations, wouldn't envoy still select the first matching route? What happens to the other rate limits in that case?

I guess I'm similarly confused by the clusters handling. I thought that the rate limits on `RateLimits.Outbound` were for each outbound service, so when we attach rate limits to the cluster destinations, do we have to match these up with the outbound service?